include/mbedtls/config_psa.h

```diff
@@ -1003,13 +1003,22 @@ extern "C" {
 #define PSA_WANT_ALG_SOME_PAKE 1
 #endif
 
-/* Temporary internal migration helpers */
-#if defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_BASIC) || \
-    defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_IMPORT) || \
-    defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_EXPORT) || \
-    defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_GENERATE) || \
-    defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE)
-#define MBEDTLS_PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_LEGACY
+/* Even though KEY_PAIR symbols' feature several level of support (BASIC, IMPORT,
+ * EXPORT, GENERATE, DERIVE) we're not planning to have support only for BASIC
+ * without IMPORT/EXPORT since these last 2 features are strongly used in tests.
+ * In general it is allowed to include more feature than what is strictly
+ * requested.
+ * As a consequence IMPORT and EXPORT features will be automatically enabled
+ * as soon as the BASIC one is. */
+#if defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_BASIC)
+#define PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_IMPORT 1
+#define PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_EXPORT 1
+#endif
+
+/* See description above */
+#if defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR_BASIC)
+#define MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR_IMPORT 1
+#define MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR_EXPORT 1
 #endif
```

Comment:
With this change, when `MBEDTLS_PSA_CRYPTO_CONFIG` is disabled, `MBEDTLS_ECP_C` no longer implies `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE`. As I've mentioned before, I think this is right because the philosophy of `MBEDTLS_PSA_CRYPTO_CONFIG` disabled is that PSA supports the same feature set as legacy, and derivation of ECC keys is not in the legacy feature set.

However, this means that in the default configuration, ECC key pair derivation is no longer available. I think this is an API break.

I see two ways to fix that:

* Make an exception to the general philosophy and have `MBEDTLS_ECP_C` enable `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE`, with a comment explaining why. Advantage: very easy to implement. Downside: makes the configuration automation more complicated to understand. Downside: larger code size for a feature that I think few users need.
* Explicitly enable `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE` in the default configuration, with a comment explaining why. This has my preference.

This is also what's causing `test_psa_compliance` to fail on the CI. We can resolve that one by explicitly enabling `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE` in `all.sh` (like [in 2.28 we enable CMAC](https://github.com/Mbed-TLS/mbedtls/blob/mbedtls-2.28.0/tests/scripts/all.sh#L3040)), if we decide that it isn't enabled in the default configuration anymore.

Reply:
If I understood correctly both options should be applied in `config_psa.h`, right?
I have some issue understanding the 2nd point though. Are you proposing to always enable `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE` when `!MBEDTLS_PSA_CRYPTO_CONFIG` no matter what it is the status of `MBEDTLS_ECP_C`?

Reply:
No: we'd make `MBEDTLS_ECP_C` imply `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_DERIVE` in addition to `PSA_WANT_KEY_TYPE_ECC_KEY_PAIR_{BASIC,IMPORT,EXPORT}`, with a note that this isn't the usual philosophy of `!MBEDTLS_PSA_CRYPTO_CONFIG` but we're doing it for backward compatibility.

Reply:
Ah ok, so I did not correctly understood the 1st comment. Sorry. I'll do as you suggested.